### ISSUE TITLE
[BUG] esphome 2024.10.0 Base MAC address from BLK3 of EFUSE version error

### ISSUE THREAD

**Issue by b2un0** [mmakaay/esphome-yeelight_bs2#130]
**Describe the bug**
after update esphome to 2024.10 i got the following errors and the lamp did not comes up and is unusable.
also the devices comes up as a "new" and no entities are shown.

```
esp-idf:000]: E (510963) system_api: Base MAC address from BLK3 of EFUSE version error, version = 74
```

**To Reproduce**
1. update esphome to 2024.10.0
2. read the logs

**Expected behavior**
a working BSLamp2 with esphome 2024.10

**Comment by randybb** [comment]
There were multiple changes related to EFUSE in esphome mentioned in this thread https://discord.com/channels/429907082951524364/1290562513565384808

Basically I am keeping all possible ignores in my config to stay on the safe side:
```yaml
esp32:
  board: esp32doit-devkit-v1
  framework:
    type: esp-idf
    sdkconfig_options:
      CONFIG_FREERTOS_UNICORE: y
      ESP_MAC_IGNORE_MAC_CRC_ERROR: y
    advanced:
      ignore_efuse_mac_crc: true
      ignore_efuse_custom_mac: true
```

**Comment by b2un0** [comment]
> There were multiple changes related to EFUSE in esphome mentioned in this thread https://discord.com/channels/429907082951524364/1290562513565384808

this link is useless without knowing the related discord server :D

**Comment by b2un0** [comment]
esphome/esphome/pull/7502
esphome/esphome-docs/pull/4297

**Comment by TomaszDom** [comment]
what do we do now? have to solder leads again to reflash all the lamps? but there's no immediate fix in the dev branch yet, correct?

**Comment by BlairC1** [comment]
mine came back online in esphome after a power cycle, working fine after adding the config from randybb.

**Comment by ezcGman** [comment]
> what do we do now? have to solder leads again to reflash all the lamps? but there's no immediate fix in the dev branch yet, correct?

you can use my branch as reference, or manually just add just this (no need for the full block, as randybb posted above):
```
esp32:
  framework:
    advanced:
      ignore_efuse_custom_mac: true
```
[EDIT]: It's actually fine to really just add this (`ignore_efuse_custom_mac`); It's doing a smart(er) merging of the configs, so no need to specify all settings under `advanced`

**Comment by randybb** [comment]
yes, ignore_efuse_mac_crc is setting ESP_MAC_IGNORE_MAC_CRC_ERROR, so this is not needed. the other ignore is only thing which you need to add to your config. After that esphome will accept MAC address stored in EFUSE.
If you have more lamps just disconnect all lamps as these are sharing now the same MAC, then power one lamp, update it from the right config, then you can power another one,...

**Comment by TomaszDom** [comment]
thanks, it worked. indeed I have >1 lamp.

pt., 18 paź 2024, 09:31 użytkownik Roman Priesol ***@***.***>
napisał:

**Comment by mmakaay** [comment]
Thanks for stepping in and helping so far @randybb ! 
I am away from home at the moment. Sunday, I'll get back and I will update the bslamp project to make the built-in config package compatible with the recent ESPHome changes.

**Comment by mmakaay** [comment]
I released version 2024.10.0 of the firmware, which addresses the reported issue.
See https://github.com/mmakaay/esphome-xiaomi_bslamp2/releases/tag/2024.10.0 for the releaes notes.

**Comment by mmakaay** [comment]
I got some reports on the new firmware working correctly with ESPHome 2024.10.0, so closing this ticket now.
